src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -552,17 +552,8 @@ public BigInteger(ulong value)
         }
 
         public BigInteger(Single value)
+            : this((Double)value)
         {
-            if (Single.IsInfinity(value))
-                throw new OverflowException(SR.Overflow_BigIntInfinity);
-            if (Single.IsNaN(value))
-                throw new OverflowException(SR.Overflow_NotANumber);
```

Comment:
As long as it does get put in a variable. There are some interesting optimisations on at least some versions, so Linqpad at least (which is not coreclr so may or may not apply, but is super-fast to check) will not produce infinity `(double)(float)(float.MaxValue * 2.0f)` as it gets optimised to a constant of `(double)float.MaxValue * 2.0` but will for `(double)(float)(float.MaxValue * 2.0)`. But if you actually assign to a `float` in the meantime then it will indeed result in infinity. Could be worth adding a test-case to cover this example though.